src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -859,21 +871,65 @@ def update_agents(cmd, client, resource_group_name, cluster_name, https_proxy=""
     # Setting the config dataplane endpoint
     config_dp_endpoint = get_config_dp_endpoint(cmd, connected_cluster.location)
 
+    if container_registry_repository is not None and default_registry is not None:
+        default_registry_error = "--default-registry and --registry-repository must not be provided at once"
+        telemetry.set_exception(exception=default_registry_error, fault_type=consts.Custom_Registry_Default_Registry_Fault_Type, summary=default_registry_error)
+        raise ArgumentUsageError(default_registry_error)
+
+    if container_registry_anonymous_pull is not None and (container_registry_username is not None or container_registry_password is not None):
+        anonymous_pull_error = "--anonymous-pull must be provided without --registry-username and --registry-password"
+        telemetry.set_exception(exception=anonymous_pull_error, fault_type=consts.Custom_Registry_Anonymous_Pull_Fault_Type, summary=anonymous_pull_error)
+        raise ArgumentUsageError(anonymous_pull_error)
+
```

Comment:
Another validation scenario we can add is when container_registry_repository is not passed but container_registry_password or container_registry_username is passed.

Reply:
Also, what is the user experience in case a user wants to move pcr from anonymous pull to password based?

Reply:
> Also, what is the user experience in case a user wants to move pcr from anonymous pull to password based?

User can simply pass username and password and it will get updated to password based pull.

Reply:
> Another validation scenario we can add is when container_registry_repository is not passed but container_registry_password or container_registry_username is passed.



> Another validation scenario we can add is when container_registry_repository is not passed but container_registry_password or container_registry_username is passed.

As this is update command, This is a valid scenario where user is not passing container_registry_repository and passing credentials. It will update the credentials to the existing PCR (we can get from helm values). For MCR case we are just ignoring these values

Reply:
> > Another validation scenario we can add is when container_registry_repository is not passed but container_registry_password or container_registry_username is passed.
> 
> > Another validation scenario we can add is when container_registry_repository is not passed but container_registry_password or container_registry_username is passed.
> 
> As this is update command, This is a valid scenario where user is not passing container_registry_repository and passing credentials. It will update the credentials to the existing PCR (we can get from helm values). For MCR case we are just ignoring these values

got it, makes sense